server/svix-server/src/expired_message_cleaner.rs

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: © 2022 Svix Authors
+// SPDX-License-Identifier: MIT
+
+use crate::db::models::message;
+use crate::error::Result;
+use chrono::Utc;
+use sea_orm::{
+    entity::prelude::*, sea_query::Expr, DatabaseConnection, DbErr, EntityTrait, UpdateResult,
+};
+use tokio::time::{sleep, Duration};
+
+/// Nullifies the payload column for expired messages
+async fn clean_expired_messages(
+    pool: &DatabaseConnection,
+) -> std::result::Result<UpdateResult, DbErr> {
+    message::Entity::update_many()
```

Comment:
It wiped my comments, but I'm still concerned about this query being grossly inefficient without an index.

Reply:
@tasn I liked your previous comments about a new model. I can move `payload` and `expiration` timestamp to a new table, and delete the whole row on expiration. This way, db scanning is limited to n = the number of unexpired messages, instead of n = the sum of all messages, and the additional read/write overhead of this feature is isolated outside the main messages table.

Reply:
I've added a partial index, based on [this discussion and answer](https://stackoverflow.com/a/31966616), that will speed up the worker query by quickly filtering out non-NULL messages.

Reply:
Yeah, it should be fine now.